datastream-server/src/test/java/com/linkedin/datastream/server/assignment/TestLoadBasedPartitionAssignmentStrategy.java

```diff
@@ -0,0 +1,164 @@
+/**
+ *  Copyright 2021 LinkedIn Corporation. All rights reserved.
+ *  Licensed under the BSD 2-Clause License. See the LICENSE file in the project root for license information.
+ *  See the NOTICE file in the project root for additional information regarding copyright ownership.
+ */
+package com.linkedin.datastream.server.assignment;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import org.apache.commons.lang3.StringUtils;
+import org.mockito.Mockito;
+import org.testng.Assert;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import com.codahale.metrics.MetricRegistry;
+
+import com.linkedin.datastream.common.Datastream;
+import com.linkedin.datastream.common.DatastreamMetadataConstants;
+import com.linkedin.datastream.common.DatastreamRuntimeException;
+import com.linkedin.datastream.common.zk.ZkClient;
+import com.linkedin.datastream.connectors.DummyConnector;
+import com.linkedin.datastream.metrics.DynamicMetricsManager;
+import com.linkedin.datastream.server.ClusterThroughputInfo;
+import com.linkedin.datastream.server.DatastreamGroup;
+import com.linkedin.datastream.server.DatastreamGroupPartitionsMetadata;
+import com.linkedin.datastream.server.DatastreamTask;
+import com.linkedin.datastream.server.DatastreamTaskImpl;
+import com.linkedin.datastream.server.PartitionThroughputInfo;
+import com.linkedin.datastream.server.providers.PartitionThroughputProvider;
+import com.linkedin.datastream.server.zk.KeyBuilder;
+import com.linkedin.datastream.server.zk.ZkAdapter;
+import com.linkedin.datastream.testutil.DatastreamTestUtils;
+import com.linkedin.datastream.testutil.EmbeddedZookeeper;
+
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+
+
+/**
+ * Tests for {@link LoadBasedPartitionAssignmentStrategy}
+ */
+@Test
+public class TestLoadBasedPartitionAssignmentStrategy {
+  private EmbeddedZookeeper _embeddedZookeeper;
+  private ZkClient _zkClient;
```

Comment:
Add tests for
a) Regular case with assignPartitions with FileBasedThroughputProvider.
b) Add tests for partition assignment when new topics get created, meaning the existing assignment is not empty and there are new topics.
c) Look at tests for StickyMulticastStrategy and add them here.

Reply:
Tests for assignment itself are in a separate file. There I already cover cases when existing assignment is not empty. I don't think there's a reason to duplicate those tests in the strategy. My goal with strategy tests was mostly to cover fallback mechanisms and zk updates.

Reply:
The following tests cases were added:

- throughput provider is not called when doing incremental assignment.
- maxTasks config is honored